From dbe78d34453245e79c42d9823f8ca4a2ebd8f76f Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
Date: Fri, 6 Jan 2023 23:11:00 +0100
Subject: Add an option to avoid setting RPATH on unix systems.

RPATH overrides the normal library search path, possibly interfering
with local policy and causing problems for multilib, among other issues.

Add an option to avoid setting it with letting it enabled by default.

Patch-Name: Add-an-option-to-avoid-setting-RPATH-on-unix-systems.patch
Signed-off-by: Sebastian Andrzej Siewior <sebastian@breakpoint.cc>
---
 CMakeLists.txt     | 4 ++--
 CMakeOptions.cmake | 3 +++
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a15ea12..cfeeb49 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -174,13 +174,13 @@ endif()
 #
 # Set RPATH for custom install prefixes
 #
-if(APPLE)
+if(APPLE AND NOT DO_NOT_SET_RPATH)
     set(CMAKE_MACOSX_RPATH 1)
 endif()
 
 include(GNUInstallDirs)
 
-if (NOT DEFINED CMAKE_INSTALL_RPATH)
+if (NOT DEFINED CMAKE_INSTALL_RPATH AND NOT DO_NOT_SET_RPATH)
     if(CMAKE_INSTALL_FULL_LIBDIR)
         set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_FULL_LIBDIR}")
     else()
diff --git a/CMakeOptions.cmake b/CMakeOptions.cmake
index d995bac..9275352 100644
--- a/CMakeOptions.cmake
+++ b/CMakeOptions.cmake
@@ -120,3 +120,6 @@ option(ENABLE_SYSTEMD
 #  Rust Targets:  https://doc.rust-lang.org/nightly/rustc/platform-support.html
 option(RUST_COMPILER_TARGET
     "Use a custom target triple to build the Rust components. Needed for cross-compiling.")
+
+option(DO_NOT_SET_RPATH
+	"Don't set the RPATH on UNIX systems.")
